dcm2niix/arrange.py

```diff
@@ -0,0 +1,260 @@
+
+    Returns:
+        dcm2niix_input_dir (str): The absolute path to the output directory containing
+            the files from the input(s).
+
+    """
+    log.info("Arrange dcm2niix input.")
+
+    if infile.endswith(".zip"):
+
+        try:
+
+            with zipfile.ZipFile(infile, "r") as zipObj:
+                log.info(f"Establishing input as zip file: {infile}")
+                is_archive_empty(zipObj)
+                dcm2niix_input_dir = extract_archive_contents(zipObj, work_dir)
+
+        except zipfile.BadZipFile:
+            log.exception(
+                (
+                    "Incorrect gear input. "
+                    "File is not a zip archive file (.zip). Exiting."
+                )
+            )
+            os.sys.exit(1)
+
+    elif infile.endswith(".tgz"):
+
+        try:
+            with tarfile.open(infile, "r") as tarObj:
+                log.info(f"Establishing input as tar file: {infile}")
+                is_archive_empty(tarObj)
+                dcm2niix_input_dir = extract_archive_contents(tarObj, work_dir)
+
+        except tarfile.ReadError:
+            log.exception(
+                (
+                    "Incorrect gear input. "
+                    "File is not a compressed tar archive file (.tgz). Exiting."
+                )
+            )
+            os.sys.exit(1)
+
+    elif rec_infile:
+        log.info(f"Establishing input as par/rec file pair: {infile} & {rec_infile}")
+        
+        # If a REC file input was provided, check infile for a valid PAR file
+        if infile.lower().endswith("par") and rec_infile.lower().endswith("rec"):
+
+            dcm2niix_input_dir, dirname = setup_dcm2niix_input_dir(infile, work_dir)
+            shutil.copy2(rec_infile, dcm2niix_input_dir)
+            shutil.copy2(infile, dcm2niix_input_dir)
+            adjust_parrec_filenames(dcm2niix_input_dir, dirname)
+
+        else:
+            log.error(
+                    (
+                        "Incorrect gear input. If rec_file_input provided, "
+                        "dcm2niix_input must be a valid PAR file. Exiting."
+                    )
+                )
+            os.sys.exit(1)
+    
+    else:
+        # Assume all other inputs will function downstream
+        dcm2niix_input_dir, dirname = setup_dcm2niix_input_dir(infile, work_dir)
+        shutil.copy2(infile, dcm2niix_input_dir)
+
+    log.info("Input for dcm2niix prepared successfully.")
+
+    return dcm2niix_input_dir
+
+
+def is_archive_empty(archiveObj):
+    """If the archive contents are empty, log an error and exit."""
+    if type(archiveObj) == zipfile.ZipFile:
+        size_contents = sum([zipinfo.file_size for zipinfo in archiveObj.filelist])
+
+    elif type(archiveObj) == tarfile.TarFile:
+        size_contents = sum([tarinfo.size for tarinfo in archiveObj.getmembers()])
+
+    else:
+        log.info(
+            "Unsupported archive format. Unable to establish size of input archive."
+        )
```

Comment:
Well, it wouldn't get to this point in the code because the function is only called if there is a zip or tar archive, so this statement is there to make the function standalone but it is not really needed. I'll add in an exit anyway.